staff/lab/ocf-tv

```diff
@@ -43,15 +43,23 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
 
 
 def volume(args):
-    call(('ssh', args.host,
-          'export PULSE_SERVER=localhost; pactl set-sink-volume {} {}%'.format(args.sink, args.amount)))
+    cmd = 'export PULSE_SERVER=localhost; '
+
+    if args.amount is not None:
+        cmd += 'pactl set-sink-volume 0 {}%; '.format(args.amount)
+
+    # thanks SO for awk tricks
+    cmd += "pactl list sinks | awk '/Sink #0/{nr[NR+9]}; NR in nr' | cut -d' ' -f 6;"
+
+    # tornado has 2 sinks, we default to sink 0 which is HDMI out to TV
+    return call(('ssh', args.host, cmd))
 
 
 def volume_in_range(val):
-    x = int(val)
+    x = int(val[1:]) if val[0] in ['+', '-'] else int(val)
     if x < 0 or x > 150:
         raise argparse.ArgumentTypeError('{} is not within the range [0, 150]'.format(val))
-    return val
+    return '-- ' + val if val[0] is '-' else val  # pactl doesn't like negatives
```

Comment:
I think he means that if you pass `pactl` an argument that begins with a negative sign, `pactl` thinks it's an option, so you need to add the `--` to make `pactl` recognize it as an argument.